lib/smack/SplitStructLoadStore.cpp

```diff
@@ -0,0 +1,137 @@
+#include "smack/SplitStructLoadStore.h"
+#include "llvm/IR/InstIterator.h"
+
+namespace smack {
+
+using namespace llvm;
+
+std::vector<Value*> getFirsts(std::vector<std::pair<Value*, unsigned>> lst) {
+    std::vector<Value*> ret;
+    for (auto p = lst.begin(); p != lst.end(); ++p)
+      ret.push_back(std::get<0>(*p));
+    return ret;
+}
+
+std::vector<unsigned> getSeconds(std::vector<std::pair<Value*, unsigned>> lst) {
+    std::vector<unsigned> ret;
+    for (auto p = lst.begin()+1; p != lst.end(); ++p)
+      ret.push_back(std::get<1>(*p));
+    return ret;
+}
+
+bool SplitStructLoadStore::runOnBasicBlock(BasicBlock& BB) {
+  std::vector<Instruction*> toRemove;
+  for(Instruction& I : BB) {
+    if (LoadInst* li = dyn_cast<LoadInst>(&I)) {
+      bool isReturnValue = false;
```

Comment:
Rewrite it like this:
http://llvm.org/docs/CodingStandards.html#turn-predicate-loops-into-predicate-functions